Libraries/Network/RCTDataManager.m

```diff
@@ -34,8 +35,37 @@ @implementation RCTDataManager
     NSMutableURLRequest *request = [NSMutableURLRequest requestWithURL:URL];
     request.HTTPMethod = [RCTConvert NSString:query[@"method"]] ?: @"GET";
     request.allHTTPHeaderFields = [RCTConvert NSDictionary:query[@"headers"]];
-    request.HTTPBody = [RCTConvert NSData:query[@"data"]];
 
+    NSString *dataURI = [RCTConvert NSString:query[@"dataURI"]];
+    NSData *data = [RCTConvert NSData:query[@"data"]];
+    if (dataURI != nil && data != nil) {
+      RCTLogError(@"You can only provide the 'data' OR 'dataURI' property on the query");
+      return;
+    }
+    if (data != nil) {
+      request.HTTPBody = data;
+      [RCTDataManager sendRequest:request responseSender:responseSender];
+      return;
+    }
+    if (dataURI != nil) {
+      if ([dataURI hasPrefix:@"assets-library"] ||
```

Comment:
At the very least we could have an `[RCTImageLoader canLoadURI:uri]` helper, so that these prefixes remain an implementation detail of `RCTImageLoader`. At some point in the future I imagine we'll have other kinds of URIs (e.g. `file:///`) with other ways of resolving them, at which point we might want to abstract resolving URIs altogether...